libwasmvm/src/cache.rs

```diff
@@ -254,19 +260,26 @@ impl From<cosmwasm_vm::AnalysisReport> for AnalysisReport {
         let cosmwasm_vm::AnalysisReport {
             has_ibc_entry_points,
             required_capabilities,
+            entrypoints,
         } = report;
 
         let required_capabilities_utf8 = set_to_csv(required_capabilities).into_bytes();
+        let entrypoints = set_to_csv(entrypoints).into_bytes();
         AnalysisReport {
             has_ibc_entry_points,
             required_capabilities: UnmanagedVector::new(Some(required_capabilities_utf8)),
+            entrypoints: UnmanagedVector::new(Some(entrypoints)),
         }
     }
 }
 
-fn set_to_csv(set: HashSet<String>) -> String {
-    let mut list: Vec<String> = set.into_iter().collect();
-    list.sort_unstable();
+/// Takes a set of string-like elements and returns a comma-separated list.
+/// Since no escaping or quoting is applied to the elements, the caller needs to ensure
+/// only simple alphanumeric values are used.
+///
+/// The order of the output elements is determined by the iteration order of the provided set.
+fn set_to_csv(set: BTreeSet<impl AsRef<str>>) -> String {
+    let list: Vec<&str> = set.iter().map(|e| e.as_ref()).collect();
     list.join(",")
```

Comment:
For the record, it seems like the intermediate vector can be avoided by using join from itertools. But since this is not on a hot path, we don't add the dependency for now.
https://stackoverflow.com/a/76223103/2013738